File: client/src/main/java/com/jraska/github/client/ChromeCustomTabsLauncher.java
package com.jraska.github.client;

import android.net.Uri;
import android.support.customtabs.CustomTabsIntent;
import com.jraska.github.client.ui.BaseActivity;
import okhttp3.HttpUrl;

import javax.inject.Provider;

final class ChromeCustomTabsLauncher implements WebLinkLauncher {
  private final Provider<BaseActivity> provider;

  ChromeCustomTabsLauncher(Provider<BaseActivity> provider) {
    this.provider = provider;
  }

  public void launch(HttpUrl url) {
    CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder().build();
    Uri uri = Uri.parse(url.toString());

    customTabsIntent.launchUrl(provider.get(), uri);
  }
}

File: client/src/main/java/com/jraska/github/client/ChromeCustomTabsLauncher.kt
package com.jraska.github.client

import android.app.Activity
import android.net.Uri
import androidx.browser.customtabs.CustomTabsIntent
import okhttp3.HttpUrl
import javax.inject.Provider

internal class ChromeCustomTabsLauncher(private val provider: Provider<Activity>) : WebLinkLauncher {
  override fun launch(url: HttpUrl) {
    val customTabsIntent = CustomTabsIntent.Builder().build()
    val uri = Uri.parse(url.toString())

    customTabsIntent.launchUrl(provider.get(), uri)
  }
}
